doc/installation.md

```diff
@@ -0,0 +1,75 @@
+# APIcast v2 installation
+
+This document explains how to install and run APIcast v2 on a clean operating system.
+
+## Installation on CentOS
+
+### Install OpenResty and dependencies
+
+OpenResty provides official pre-built packages for CentOS, the latest installation instructions are provided in [OpenResty documentation](https://openresty.org/en/linux-packages.html).
+
+You will need to create a file `/etc/yum.repos.d/OpenResty.repo` with the following content:
+
+```
+[openresty]
+name=Official OpenResty Repository
+baseurl=https://copr-be.cloud.fedoraproject.org/results/openresty/openresty/epel-$releasever-$basearch/
+skip_if_unavailable=True
+gpgcheck=1
+gpgkey=https://copr-be.cloud.fedoraproject.org/results/openresty/openresty/pubkey.gpg
+enabled=1
+enabled_metadata=1
+```
+
+Install OpenResty and the `resty` command-line utility, which is required for APIcast v2:
+
+```shell
+sudo yum install openresty openresty-resty
+```
+
+APIcast v2 uses LuaRocks for managing Lua dependencies. As it is not in the standard Yum repositories, you must first enable the [EPEL](https://fedoraproject.org/wiki/EPEL) (Extra Packages for Enterprise Linux) package repository with the following command:
+
+```shell
+sudo yum install epel-release
+```
+
+Install LuaRocks:
+```shell
+sudo yum install luarocks
+```
+
+If you are using OAuth authentication method, you will also need to install Redis. This can be done with `sudo yum install redis` command (note that you will need the [EPEL](https://fedoraproject.org/wiki/EPEL) for this).
```

Comment:
As it says, you will need it "if you are using OAuth authentication method" - the only currently supported OAuth flow (authorization code flow, where nginx generates tokens) requires Redis.